w3af/plugins/grep/cdn_providers.py

```diff
@@ -0,0 +1,255 @@
+        [".yottaa.net", "Yottaa"],
+        [".cubecdn.net", "cubeCDN"],
+        [".cdn77.net", "CDN77"],
+        [".cdn77.org", "CDN77"],
+        [".incapdns.net", "Incapsula"],
+        [".bitgravity.com", "BitGravity"],
+        [".r.worldcdn.net", "OnApp"],
+        [".r.worldssl.net", "OnApp"],
+        ["tbcdn.cn", "Taobao"],
+        [".taobaocdn.com", "Taobao"],
+        [".ngenix.net", "NGENIX"],
+        [".pagerain.net", "PageRain"],
+        [".ccgslb.com", "ChinaCache"],
+        ["cdn.sfr.net", "SFR"],
+        [".azioncdn.net", "Azion"],
+        [".azioncdn.com", "Azion"],
+        [".azion.net", "Azion"],
+        [".cdncloud.net.au", "MediaCloud"],
+        [".rncdn1.com", "Reflected Networks"],
+        [".cdnsun.net", "CDNsun"],
+        [".mncdn.com", "Medianova"],
+        [".mncdn.net", "Medianova"],
+        [".mncdn.org", "Medianova"],
+        ["cdn.jsdelivr.net", "jsDelivr"],
+        [".nyiftw.net", "NYI FTW"],
+        [".nyiftw.com", "NYI FTW"],
+        [".resrc.it", "ReSRC.it"],
+        [".zenedge.net", "Zenedge"],
+        [".lswcdn.net", "LeaseWeb CDN"],
+        [".lswcdn.eu", "LeaseWeb CDN"],
+        [".revcn.net", "Rev Software"],
+        [".revdn.net", "Rev Software"],
+        [".caspowa.com", "Caspowa"],
+        [".twimg.com", "Twitter"],
+        [".facebook.com", "Facebook"],
+        [".facebook.net", "Facebook"],
+        [".fbcdn.net", "Facebook"],
+        [".cdninstagram.com", "Facebook"],
+        [".rlcdn.com", "Reapleaf"],
+        [".wp.com", "WordPress"],
+        [".aads1.net", "Aryaka"],
+        [".aads-cn.net", "Aryaka"],
+        [".aads-cng.net", "Aryaka"],
+        [".squixa.net", "section.io"],
+        [".bisongrid.net", "Bison Grid"],
+        [".cdn.gocache.net", "GoCache"],
+        [".hiberniacdn.com", "HiberniaCDN"],
+        [".cdntel.net", "Telenor"],
+        [".raxcdn.com", "Rackspace"],
+        [".unicorncdn.net", "UnicornCDN"],
+        [".optimalcdn.com", "Optimal CDN"],
+        [".kinxcdn.com", "KINX CDN"],
+        [".kinxcdn.net", "KINX CDN"],
+        [".stackpathdns.com", "StackPath"],
+        [".hosting4cdn.com", "Hosting4CDN"],
+        [".netlify.com", "Netlify"],
+        [".b-cdn.net", "BunnyCDN"],
+        [".pix-cdn.org", "Advanced Hosters CDN"],
+        [".roast.io", "Roast.io"],
+        [".cdnvideo.ru", "CDNvideo"],
+        [".cdnvideo.net", "CDNvideo"],
+        [".trbcdn.ru", "TRBCDN"],
+        [".cedexis.net", "Cedexis"],
+        [".streamprovider.net", "Rocket CDN"],
+    ]
+
+    def grep(self, request, response):
+        """
+        Check if responses are hosted by CDN provider
+        """
+        headers = response.get_headers()
+        for cdn_header in self.cdn_headers:
+            cdn_header_name = cdn_header[0]
+            if cdn_header_name in headers:
+                if cdn_header[1] == headers[cdn_header_name]:
+                    self._detect_cdn(response, cdn_header[2])
+
+        url = request.get_url()
```

Comment:
Do you want to run URL detection even after the CDN was identified using the headers? This might yield duplicates in some cases?

Also, for a very very small % of requests, the ones that match to a CDN by domain; the `grep` method will be faster if you do the URL match first.

Reply:
You're right, I thought about case when plugin detects another CDN provider with header and another with URL. But in fact if someone adds wrong CDN info to header.. It just shouldn't happen. I changed it to return after detecting provider via header. URL is here as backup option if we can't detect CDN with header.